engine/gamesys/src/dmsdk/gamesys/resources/res_texture.h

```diff
@@ -22,6 +22,10 @@ namespace dmGameSystem
     struct TextureResource
     {
         dmGraphics::HTexture m_Texture;
+        uint8_t              m_Uploading:1;
+        uint8_t              m_DelayDelete:1;
```

Comment:
I guess these can't both be one, right? Maybe the resource should have it's own state param instead

Reply:
Well, we do set the delay delete because it is currently uploading.

Reply:
That said, it would be nice if the resource system had some extra scaffolding around this concept.